main.c

```diff
@@ -431,6 +434,43 @@ uint32_t parse_color(const char *color) {
 	return res;
 }
 
+static void format_result(const char *format, char *buffer, const int buflen, struct slurp_box result) {
```

Comment:
I'd prefer the first argument to be `result`, as the function is named `format_result`. Also, `result` could be a `const struct slurp_box *`.